lib/regressions/multivariableLinearRegression.ts

```diff
@@ -0,0 +1,75 @@
+import { meanSquaredError } from '../functions/losses'
+import { updateWeightsAndBias } from '../functions/optimizers'
+
+export function predictionMultivariable(input: number[][], weight: number[], bias: number): number[] {
+    const value: number[] = []
+
+    input.forEach((feature) => {
+        let pred = 0
+        for (let i = 0; i < input[0].length; i++) {
+            pred += feature[i] * weight[i]
+        }
+        value.push(pred + bias)
+    })
+
+    return value
+}
+
+export default class MultivariableLinearRegression {
+    static *fit(
+        input: number[][],
+        target: number[],
+        weight: number[],
+        bias: number,
+        learningRate: number,
+        epochs: number,
+        costFunction: Function
+    ): Generator<{ updatedWeight: number[]; updatedBias: number; costHistory: number[] }> {
```

Comment:
you can export this as a type:
`export type MultivariableLienarRegressionOutputType = { updatedWeight: number[]; updatedBias: number; costHistory: number[] }` (or a smart name)
and use it here as `Generator<MultivariableLienarRegressionOutputType>`